18_regexp_lab/01.py

```diff
@@ -9,4 +9,22 @@
 Write a program that takes a property file name and key
 as command line arguments and prints the requested value
 """
+import re
+import argparse
 
+parser = argparse.ArgumentParser()
+parser.add_argument("file", type=str, help="The file you want to scan")
+parser.add_argument("key", type=str, help="sreach of key value")
+args = parser.parse_args()
+
+
+def getvaule(key, f):
+    #(?<=key = ).*
+    res = re.search(r"(?<=" + key + r' = ).*', f)
+    if res is not None:
+        return res.group(0)
+    else: return "key not found"
+
+with open(args.file, "r") as f:
+    print getvaule(args.key, f.read())
+    
```

Comment:
Nice first version but it has a number of problems. e.g. if there is a line
```
# name = bob
```
and we are looking for the 'name' field?
or if the number of spaces around the = signe is not 1 and 1 then this regex won't match.

In general I'd probably first read in the whole file and fill a dictionary with the key-value pairs and then use that to look-up the requested key.